File: garden advice/garden-app/garden_advice.py
"""
Garden Advice Program
This program provides gardening advice based on the season and plant type entered by the user.
It also recommends plants suitable for the current season.
"""

def get_advice(season, plant_type):
    """
    Generates gardening advice based on season and plant type.
    
    Args:
        season (str): The current season (e.g., 'summer', 'winter').
        plant_type (str): The type of plant (e.g., 'flower', 'vegetable').
        
    Returns:
        str: A string containing the advice.
    """
    
    # Dictionary storing advice for seasons
    seasonal_advice = {
        "summer": "Water your plants regularly and provide some shade.",
        "winter": "Protect your plants from frost with covers.",
        "spring": "Prune dead branches and add compost to the soil.",
        "autumn": "Clean up fallen leaves and plant bulbs for spring."
    }
    
    # Dictionary storing advice for plant types
    plant_advice = {
        "flower": "Use fertiliser to encourage blooms.",
        "vegetable": "Keep an eye out for pests!",
        "herb": "Harvest frequently to encourage growth.",
        "tree": "Ensure deep watering for strong root systems."
    }
    
    advice = []
    
    # Get seasonal advice
    advice.append(seasonal_advice.get(season.lower(), "No specific advice for this season."))
    
    # Get plant type advice
    advice.append(plant_advice.get(plant_type.lower(), "No specific advice for this plant type."))
    
    return "\n".join(advice)

def recommend_plants(season):
    """
    Recommends plants based on the season.
    
    Args:
        season (str): The current season.
        
    Returns:
        str: A recommendation string.
    """
    recommendations = {
        "summer": "Sunflowers, Zinnias, Tomatoes, Peppers",
        "winter": "Pansies, Kale, Spinach, Garlic",
        "spring": "Tulips, Daffodils, Lettuce, Peas",
        "autumn": "Chrysanthemums, Asters, Carrots, Broccoli"
    }
    
    return recommendations.get(season.lower(), "No specific recommendations for this season.")

def main():
    """
    Main function to run the program.
    """
    print("Welcome to the Garden Advice Program!")
    
    # User interaction: Get input for season and plant type
    season = input("Enter the current season (summer, winter, spring, autumn): ").strip().lower()
    plant_type = input("Enter the plant type (flower, vegetable, herb, tree): ").strip().lower()
    
    print("\n--- Gardening Advice ---")
    advice = get_advice(season, plant_type)
    print(advice)
    
    print("\n--- Plant Recommendations ---")
    recommendation = recommend_plants(season)
    print(f"Recommended plants for {season}: {recommendation}")

if __name__ == "__main__":
    main()

File: garden advice/garden_advice.js
// Garden Advice Logic

function getAdvice(season, plantType) {
    const seasonalAdvice = {
        "summer": "Water your plants regularly and provide some shade.",
        "winter": "Protect your plants from frost with covers.",
        "spring": "Prune dead branches and add compost to the soil.",
        "autumn": "Clean up fallen leaves and plant bulbs for spring."
    };

    const plantAdvice = {
        "flower": "Use fertiliser to encourage blooms.",
        "vegetable": "Keep an eye out for pests!",
        "herb": "Harvest frequently to encourage growth.",
        "tree": "Ensure deep watering for strong root systems."
    };

    const sAdvice = seasonalAdvice[season] || "No specific advice for this season.";
    const pAdvice = plantAdvice[plantType] || "No specific advice for this plant type.";

    return `${sAdvice}\n${pAdvice}`;
}

function recommendPlants(season) {
    const recommendations = {
        "summer": ["Sunflowers", "Zinnias", "Tomatoes", "Peppers"],
        "winter": ["Pansies", "Kale", "Spinach", "Garlic"],
        "spring": ["Tulips", "Daffodils", "Lettuce", "Peas"],
        "autumn": ["Chrysanthemums", "Asters", "Carrots", "Broccoli"]
    };

    const plants = recommendations[season] || [];
    
    if (plants.length > 0) {
        return plants.join(", ");
    } else {
        return "No specific recommendations available for this season.";
    }
}

// Function called by the button in HTML
function provideAdvice() {
    // 1. Get input values
    const seasonInput = document.getElementById("season").value;
    const plantTypeInput = document.getElementById("plantType").value;

    // 2. Generate content
    const adviceText = getAdvice(seasonInput, plantTypeInput);
    const recommendationText = recommendPlants(seasonInput);

    // 3. Display results
    document.getElementById("adviceOutput").textContent = adviceText;
    document.getElementById("recommendationOutput").textContent = recommendationText;
    
    // Show the results container
    document.getElementById("results").style.display = "block";
}
